packages/atomic/src/components/commerce/product-template-components/atomic-product-image/atomic-product-image.tsx

```diff
@@ -0,0 +1,191 @@
+   */
+  @Method() public async navigateToImage(index: number) {
+    this.currentImage = index;
+  }
+
+  private logWarning(message: string) {
+    this.bindings.engine.logger.warn(message, this.host);
+  }
+
+  private handleImageError(event: Event) {
+    const image = event.target as HTMLImageElement;
+    const message = `The image url "${image.src}" is not valid or could not be loaded. You might want to add a "fallback" property.`;
+    if (this.fallback && image.src !== this.fallback) {
+      this.useFallback = true;
+      image.src = this.fallback;
+    } else if (image.src === this.fallback) {
+      this.logWarning('The fallback image failed to load.');
+    } else {
+      this.logWarning(message);
+    }
+  }
+
+  private handleMissingFallback(message: string) {
+    if (!this.fallback) {
+      this.logWarning(message);
+      this.host.remove();
+      return null;
+    }
+    return this.fallback;
+  }
+
+  private validateUrl(url: string | undefined) {
+    if (!url) {
+      const message = `Image for ${this.product.ec_name} is missing. Please review your indexation. You might want to add a "fallback" property.`;
+      return this.handleMissingFallback(message);
+    }
+
+    if (typeof url !== 'string') {
+      const message = `Expected "${this.field}" to be a text field. Please review your indexation. You might want to add a "fallback" property.`;
+      return this.handleMissingFallback(message);
+    }
+
+    return url;
+  }
+
+  componentWillLoad(): void {
+    const validImages = this.imageUrls.filter(
+      (image) => typeof image === 'string'
+    );
+
+    this.images = validImages.map((url, index) => {
+      const finalUrl = this.useFallback ? this.fallback : url;
+
+      this.validateUrl(finalUrl);
+
+      return {
+        src: filterProtocol(finalUrl),
+        alt: `Image ${index + 1} out of ${validImages?.length} for ${this.product.ec_name}`,
+      };
+    }) as Image[];
+  }
+
+  private get imageUrls() {
+    const value = ProductTemplatesHelpers.getProductProperty(
+      this.product,
+      this.field
+    );
+
+    return Array.isArray(value) ? value : [value];
+  }
+
+  private get numberOfImages() {
+    return this.images?.length ?? 0;
+  }
+
+  private renderCurrentImage(image: Image | null) {
+    if (image === null) {
+      return;
+    }
+
+    return (
+      <img
+        class="aspect-square"
+        alt={image.alt}
+        src={image.src}
+        onError={(image) => this.handleImageError(image)}
+        loading="lazy"
+      />
+    );
+  }
+
+  public render() {
+    const imagesToRender = this.images.map((image: Image, index: number) => {
+      return {
+        src: image.src,
+        alt: `Image ${index} out of ${this.images?.length} for ${this.product.ec_name}`,
+      };
+    });
+    if (this.images.length === 0) {
+      this.validateUrl(this.fallback);
+      return (
+        <img alt={'No image available'} src={this.fallback} loading="eager" />
```

Comment:
`No image available` should ideally be translated/loaded from i18n.

Reply:
What about the "image x out of x for product x" ?

Reply:
SHould come from a field as discussed. In a future PR, it'll be interesting to back-port that (the alt stuff) to the other atomic-image in the search-search (not-commerce-search) use case

Reply:
https://coveord.atlassian.net/browse/KIT-3180